internal/runtime/runtime_test.go

```diff
@@ -124,7 +128,11 @@ func TestProcessMessageOneCustomTransform(t *testing.T) {
 	}
 	runtime := NewGolangRuntime("", nil, dic)
 	runtime.SetDefaultFunctionsPipeline([]interfaces.AppFunction{transform1})
-	result := runtime.ProcessMessage(context, envelope, runtime.GetDefaultPipeline())
+
+	messageData, decodeError, _ := runtime.DecodeMessage(context, envelope)
+	require.Nil(t, decodeError)
+
+	result := runtime.ProcessMessage(context, messageData, runtime.GetDefaultPipeline())
```

Comment:
maybe worth adding `require.NotNil(t, messageData)` for these (and subsequent calls)?

Reply:
Good catch!

Reply:
added